kornia/augmentation/random_generator/base.py

```diff
@@ -37,3 +38,57 @@ def make_samplers(self, device: torch.device, dtype: torch.dtype) -> None:
 
     def forward(self, batch_shape: torch.Size, same_on_batch: bool = False) -> Dict[str, torch.Tensor]:  # type:ignore
         raise NotImplementedError
+
+
+class DistributionWithMapper(Distribution):
+    """Wraps a distribution with a value mapper.
+
+    This is used to restrict the output values of a given distribution by a value mapper.
+    The value mapper can be functions like sigmoid, tanh, etc.
+
+    Args:
+        dist: the target distribution.
+        mapper: the function or module to adjust the output from distributions.
+
+    Example:
+        >>> from torch.distributions import Normal
+        >>> import torch.nn as nn
+        >>> # without mapper
+        >>> dist = DistributionWithMapper(Normal(0., 1.,), mapper=None)
+        >>> _ = torch.manual_seed(0)
+        >>> dist.rsample((8,))
+        tensor([ 1.5410, -0.2934, -2.1788,  0.5684, -1.0845, -1.3986,  0.4033,  0.8380])
+        >>> # with sigmoid mapper
+        >>> dist = DistributionWithMapper(Normal(0., 1.,), mapper=nn.Sigmoid())
```

Comment:
just to get it right -- this applies Normal and then sigmoid ?

Reply:
Yes. I will rename it to value_mapper to make it more clear.